uncompyle6/semantics/customize36.py

```diff
@@ -148,61 +165,21 @@ def call_ex_kw2(node):
         """Handle CALL_FUNCTION_EX 2  (have KW) but with
         BUILD_{MAP,TUPLE}_UNPACK_WITH_CALL"""
 
-        # This is weird shit. Thanks Python!
-        self.preorder(node[0])
-        self.write('(')
-
-        assert node[1] == 'build_tuple_unpack_with_call'
-        btuwc = node[1]
-        tup = btuwc[0]
-        if tup == 'expr':
-            tup = tup[0]
+        self.template_engine(
```

Comment:
When a function has a single call to `self.template engine()`, followed by a call to `self.prune()` there is a shorthand for this by adding it into `TABLE_DIRECT` for example:
```
'call_ex_kw2': 
            ("%c(%p, %p)",
            (0, 'expr'), (1, 'build_tuple_unpack_with_call', 100),
            (2, 'build_map_unpack_with_call', 100)), node)
```

However the reason I didn't do that was probably because of handling the positional args like we do now in `call_ex_kw` now, thanks to you. 

So let's abstract the handling of the formatting of the positional arg like is done now in `call_ex_kw` and use that subroutine in `call_ex_kw2` now as well. What do you think?

Reply:
Yes, You are right, Please let me adjust as you said and retry.

Reply:
@rocky, there is one questions, in this commit i prepare `build_tuple_unpack_with_call` in the way it handles positional args formatting as we implemented before (For special case `LOAD_CONST`). the only thing is left if the result is empty we should omit `,`.  
besides if i get the concept right, in `call_ex_kw2` we have `build_tuple_unpack_with_call` but in `call_ex_kw` we don't have such an opr.  is it still needed to use our formatter? or possibly invoke `build_tuple_unpack_with_call` manually in `call_ex_kw2`?

Sorry for these questions. there's still a long way for me to understand all the concepts.
Thank you

Reply:
I am not sure I know which is best. I leave it up to you. 

If there isn't much in the way of formatting, you do not have to call the formatter `template_engine()` . You can instead call `build_tuple_unpack_with_call()` directly if that is easier and clearer. 

I find all of this very complicated too. And that is also why the code is inconsistent and still a mess. 

If you are up to it after this 3.6 format strings is badly done. I have started code for that but haven't been able to finish it. Here's grammar rules really need to change.

Reply:
I see. i'll update this PR.

> If you are up to it after this 3.6 format strings is badly done. I have started code for that but haven't been able to finish it. Here's grammar rules really need to change.

 I prefer to help in order of your priorities. Say 3.6 fmt or whatever you decide.